### DISCUSSION TITLE
Where does ext_loader loads functions from?

### DISCUSSION THREAD

**Discussion by Divelix** [open-mmlab/mmcv#2628] [Q&A] [answered]
Currently, I'm trying to extract PointPillars from mmdetection3d (to deploy on edge device) and stuck on [voxelization step](https://github.com/open-mmlab/mmcv/blob/master/mmcv/ops/voxelize.py). As I understand, this operation should be written in CUDA somewhere, but I see only `ext_loader` thing that pulls `dynamic_voxelize_forward` and `hard_voxelize_forward` functions from somewhere. Can someone clarify, how `ext_loader` works and where can I find latest implementation of these ops?

**Comment by enesozi** [comment, marked as answer]
@Divelix You may find the cpu version [here](https://github.com/open-mmlab/mmcv/blob/master/mmcv/ops/csrc/pytorch/cpu/voxelization.cpp#L6). There's also a cuda implementation that you may check out.
